arviz/stats/diagnostics.py

```diff
@@ -65,7 +65,9 @@ def bfmi(data):
     return _bfmi(dataset.energy)
 
 
-def ess(data, *, var_names=None, method="bulk", relative=False, prob=None):
+def ess(
```

Comment:
Instead of setting kwargs can we use decorators that signify dask? Otherwise kwargs can be anything and itll be hard to trace whats going on

Reply:
Btw I know Oriol said use **kwargs so this is a slightly different change in direction but since this is a major architectural change want to explore a couple options before we settle on one, and after seeing this one realize some of the downsides

Reply:
I see. So for that decorator we could have `dask.enable(**kwargs)` and `dask.disable` in the `dask` class to control just like numba and also take dask kwargs from user to use for the decorator?

Reply:
That would be ideal because its much more obvious, but its just an idea I dont know for sure if it will work. Would welcome feedback on the feasability

Reply:
I've added a sample class and decorator for `wrap_xarray_ufunc`. This is just to see if this is what we want :)